WalletWasabi.Tests/StoreTests/AsyncMutexTests.cs

```diff
@@ -0,0 +1,171 @@
+using Nito.AsyncEx;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace WalletWasabi.Tests.StoreTests
+{
+	public class AsyncMutexTests
+	{
+		[Fact]
+		public async Task AsyncMutexTests1Async()
+		{
+			AsyncMutex asyncMutex = new AsyncMutex("mutex1");
+
+			// Cannot be IDisposable because the pattern is like Nito's AsyncLock.
+			Assert.False(asyncMutex is IDisposable);
+
+			// Use the mutex two times after each other.
+			using (await asyncMutex.LockAsync())
+			{
+				await Task.Delay(1);
+			}
+
+			using (await asyncMutex.LockAsync())
+			{
+				await Task.Delay(1);
+			}
+
+			// Release the Mutex from another thread.
+
+			var disposable = await asyncMutex.LockAsync();
+
+			var myThread = new Thread(new ThreadStart(() =>
+			{
+				disposable.Dispose();
+			}));
+			myThread.Start();
+			myThread.Join();
+
+			using (await asyncMutex.LockAsync())
+			{
+				await Task.Delay(1);
+			}
+
+			// Acquire the Mutex with a background thread.
+
+			var myTask = Task.Run(async () =>
+			{
+				using (await asyncMutex.LockAsync())
+				{
+					await Task.Delay(3000);
+				}
+			});
+
+			// Wait for the Task.Run to Acquire the Mutex.
+			await Task.Delay(100);
+
+			// Try to get the Mutex and save the time.
+			DateTime timeOfstart = DateTime.Now;
+			DateTime timeOfAcquired = default;
```

Comment:
Why not name the variables `startedTime` and `acquiredTime`

Reply:
It was copypaste, I just moved the files, but anyhow, I can see arguments for both naming scheme.